README.md

```diff
@@ -1,26 +1,151 @@
 # Stencil Icon
 
-Stencil Icon Component
+A container for displaying SVG icons from a `<symbol>`-based SVG sprite.
 
-[Link to demo](#)
+**View demo (coming soon)**
 
 ## Requirements
 
-- Sass
-- AMD loader
-- Dust templating
+- AdaptiveJS 2.1 or greater
 
 ## Installation
 
-Installation instructions. Be as brief as possible without leaving out necessary requirements.
+```shell
+cd my/adaptive/project
+grunt component:install:icon
+```
 
 ## Usage
 
-- How to
+The Icon component ships with a default icon set, [Shoppicon](http://github.com/ry5n/shoppicon). Usage depends in whether you want to use the default set, or a custom set.
 
-## Development
+### Default icon set
 
-* run `npm install`
-* run `bower install`
-* run `grunt serve`
-* navigate to [localhost:3000/tests/visual](http://localhost:3000/tests/visual)
+1. In your adaptive project, open `app/global/base-view.js`, and add the following dependency:
+
+    ```javascript
+    define(['$', ..., 'text!bower_components/shoppicon/svg-sprite/shoppicon.svg'], function($, ..., icons) {...});
+    ```
+
+    Then, within the views’s returned context, add a key for the icon sprite:
+
+    ```javascript
+    return {
+        preProcess: function(context) {...},
+        postProcess: function(context) {...},
+        context: {
+            ...
+            icons: icons,
+        },
+    };
+    ```
+
+    Finally, within `app/global/base.dust`, include the svg sprite *at the very top of the body tag*, like so:
+
+    ```html
+    {body|openTag|s}
+        <div hidden>{icons|s}</div>
+        
+        {+bodyBlock}
+        ...
+        {/bodyBlock}
+    </body>
+    ```
+
+2. Use the Icon component’s dust helper to render icons. For example, to show a hamburger icon:
+
+    ```html
+    {@c-icon set="shoppicon" name="menu" /}
+    ```
+
+
+### Custom icon set
+
+1. To use a custom icon set, you’ll need to build the SVG sprite using a separate tool, such as (grunt-svgstore)[https://github.com/FWeinb/grunt-svgstore], or (grunt-iconpack)[https://github.com/ry5n/grunt-iconpack]. Preferably, build your sprite with ids that are prefixed with `icon-`.
+
+2. Follow step 2 in “Default icon set” above, replacing the path to the shoppicon sprite with the path to your custom sprite.
+
+3. Use the Icon component’s dust helper to render icons. If your prefix is set correctly, you don’t need the `set` parameter. For example, to show a hamburger icon:
+
+    ```html
+    {@c-icon name="menu" /}
+    ```
+
+### Changing the icon dynamically
+
+To update the icon’s displayed symbol from script, you need to enable dynamic mode in the dust helper:
+
+```html
+{@c-icon name="menu" dynamic="true" /}
```

Comment:
This is interesting, and the performance optimization is nice. However, I'm not sure I'm a fan that you have to remember to enable dynamic mode in order to interact with this component via JS.

Is there a way have this optimization without requiring developers to remember to add an attribute?